### ISSUE TITLE
Time picker default value in Mac OS

### ISSUE THREAD

**Issue by capi20** [mui/mui-x#11519]
### Steps to reproduce

Link to live example: (required)

Steps:
1. Open time picker on Mac device
2. Run below code and pass variable time to value prop in time picker
  let date = new Date()
  let currDate = date.toLocaleDateString() // this will give today's date
  let time = new Date(`${currDate} 03:00 AM`)
3. On first load time picker is throwing error

### Current behavior

I have time in string format ("03:00 AM") which I need to display in time picker. For that first I am creating a date object by below code
let date = new Date()
let currDate = date.toLocaleDateString() // this will give today's date
let time = new Date(`${currDate} 03:00 AM`)

After creating date object, passing variable time to value prop in time picker. On windows devices, Time picker is rendering correct time without any error but on Mac devices (checked on safari and chrome both), it's throwing invalid time error.

### Expected behavior

On Mac OS also time picker should render correct time without any error.

### Context

Implementing time picker in an application 

### Your environment

  Binaries:
    Node: 14.16.1
    npm: 6.14.12
  Browsers:
    Chrome: Not Found
    Safari: Not found

**Search keywords**: time picker

**Comment by LukasTy** [comment]
Hello @capi20
Please consider using a solution that would not rely on localization to work.
Your approach might work in a certain locale, where `toLocaleDateString` would produce a date in the ISO string.

I take it you are using `date-fns` library if you are using `Date` for your dates, then you can use `parse` to correctly handle Date regardless of the user's locale.
Here is a working [codesandbox example](https://codesandbox.io/p/sandbox/flamboyant-bogdan-6tlv2x?file=%2Fsrc%2FDemo.tsx%3A10%2C44).

TL/DR: The problem is not related to OS or browser, but the incorrect parsing of date, which depends on certain locale.
